### PR TITLE
Playwright: update e2e documentation.

### LINKED ISSUE #53170: Write up Playwright code style.
**Task Details**
There is enough difference between the Playwright and Selenium frameworks that I think an update to the code style (including naming variables, etc) should be updated.

**Related to**
Parent: #52464

### LINKED ISSUE #57149: [Meta] Playwright: Update documentation
### Details

This is a big grouper issue for ideas to ensure are present in documentation and other updates to add to documentation re: Playwright.

### Checklist

- [ ] Update the Conifg FG page to give `cd test/e2e` command, and setting values in `.zshrc`
- [ ] Draw more intention to using `page.pause` and the Playwright inspector
- [ ] Using Playwright traces
- [ ] Which attribute to add to HTML elements if you need to make selection easier
- [ ] Async navigation
- [ ] Waiting on a specific URL when navigating to handle the multi-redirects we do
- [ ] `waitForSelector` as a means of validation
- [ ] Handlers not being attached yet to elements, and waiting on other keying elements that indicate buttons/links should have handlers

### Related

_No response_

### LINKED ISSUE #58370: Playwright: update GitHub documentation
### Details

Update GitHub documentation for e2e tests.

### Checklist

- [ ] make a 'getting started' page.
- [ ] revise style guide.
- [ ] consider a tool like readthedocs.

### Related

Child of #57149.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR overhauls the documentation for the E2E tests.

Key changes:
- removal of all references to Selenium.
- restructure top level files (Setup, Running Tests, Library Objects).
- update examples.
- include animated GIFs and images for reference.

#### Testing instructions

Go to [this page](https://github.com/Automattic/wp-calypso/blob/update/e2e-documentation/test/e2e/README.md) and start following along the documentation.

Closes #58370.

### CODE REVIEW

**Comment by worldomonation** [comment]
@cometgrrl Do you have any more suggestions and/or changes you'd like to see?

File: test/e2e/docs/tests_local.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+```
+
+## Running tests
+
+### Individual spec file(s)
+
+Specify the file(s) directly:
+
+```
+yarn jest specs/specs-playwright/wp-support__popover.ts specs/specs-playwright/wp-support__home.ts
```
Comment (cometgrrl) [inline test/e2e/docs/tests_local.md]: not a big deal, but neither of these specs exist with this name, so if you copy/paste and try to run, it doesn't find any tests to run.

File: test/e2e/docs/tests_local.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+5. ensure requests to `http://calypso.localhost:3000` are registering in your local instance.
+
+The local environment is now ready for testing. When a test is run, it will hit the local development server instead of the WordPress.com staging environment.
+
+### Debug mode
+
+Refer to the [Debugging](debugging.md) page for techniques on running a test in debug mode.
+
+#0## Notes on TypeScript
```
Comment (cometgrrl) [inline test/e2e/docs/tests_local.md]: Is this supposed to be ### (or ####) and not #0##?

File: test/e2e/docs/library_objects.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
+# Style Guide
```
Comment (cometgrrl) [inline test/e2e/docs/library_objects.md]: Should this be #Library Objects?

**Review by cometgrrl — APPROVED** [review]
Overall, it looks great. Just a few small comments, that you can address or not as you wish.

File: test/e2e/docs/tests_local.md (hunk 1 above)
Comment (worldomonation) [reply test/e2e/docs/tests_local.md]: Gotcha, they _used_ to exist but have been renamed/moved. Will update references here.

File: test/e2e/docs/tests_local.md (hunk 2 above)
Comment (worldomonation) [reply test/e2e/docs/tests_local.md]: Good catch!

File: test/e2e/docs/library_objects.md (hunk 3 above)
Comment (worldomonation) [reply test/e2e/docs/library_objects.md]: Yes 😓

**Pull request merged**